### PR TITLE
Fix django authorization redirect by correctly checking validity of credentials

### PR DESCRIPTION
**Pull request by waprin**
I was checking for credentials but I need to use the `has_credentials` method for cases of existing but invalid credentials.

### CODE REVIEW

**Review by theacodes — APPROVED** [review]

**Comment by jonparrott** [comment]
@waprin looks like we have a coverage miss:

```
oauth2client/contrib/django_util/views.py              73      1     14      1    98%   187, 186->187
```

Can you fix?

**Comment by waprin** [comment]
@jonparrott Looks like all my tests pass and django_util now has 100% coverage, failing on cover because of other code I think?

**Comment by waprin** [comment]
Ah never mind, now I see still some missing lines in the tests themselves

File: oauth2client/contrib/django_util/views.py [hunk 1]
```diff
@@ -170,21 +170,22 @@ def oauth2_authorize(request):
          A redirect to Google OAuth2 Authorization.
     """
     return_url = request.GET.get('return_url', None)
+    if not return_url:
```
Comment (nathanielmanistaatgoogle) [inline oauth2client/contrib/django_util/views.py:173]: Prefer `is None`/`is not None` when comparing to the `None` object.

If `request.GET.get('return_url', None)` is the empty string, that's a bug, right? And by unintentionally sweeping it along the same code path as `request.GET`-does-not-contain-a-key-value-mapping-for-key-`return_url` we make finding and fixing the bug harder, right?

**Pull request merged**